scripts/configurator/parser/cmake_row.py

```diff
@@ -87,18 +62,16 @@ def is_pound_comment(self, row: str) -> bool:
       return row[0] == "#"
 
    def serialize(self) -> str:
-      serialized_val: str
       if self.row_type == row_type.VARIABLE:
-         if self.configurator_type == configurator_type.BOOL:
-            serialized_val = "ON" if self.val else "OFF"
-         else:
-            serialized_val = str(self.val)
-         return self.name + ":" + self._cmake_type_str + "="  + serialized_val
-      elif self.row_type == row_type.POUND_COMMENT:
+         return self.name + ":" + self._cmake_type + "=" + self.val.serialize()
+      elif type(self.val) == str and self.row_type == row_type.POUND_COMMENT:
```

Comment:
No need to check for `type(self.val) == str` here. This is an invariant of your code and always must hold when the row_type is a comment. Therefore, just check for `self.row_type == row_type.POUND_COMMENT` and, if you want, just `assert(type(self.val) == str)` inside the `if`. With `if` statements we should check for things that are expected to be true or false. Invariants are checked with asserts, instead.

Also, the whole serialize() should be conceptually implemented with a big switch statement. Because python doesn't have that, a sequence of if/elif will do the job.

Reply:
This is just needed for mypy to stop complaining. The invariant holds, but unfortunately mypy doesn't like it.

Reply:
OK, then use `str(self.val)` instead of this ugly checking.